Add Chinese translation for models - Part V

ZhengZhenyu opened the pull request:
Add Chinese translation for SimpleLicensing and Software models, this is the last patch for the series.

ZhengZhenyu committed:
Add Chinese translation for models - Part V

Add Chinese translation for SimpleLicensing and Software models, this is the last patch for the series.

Signed-off-by: Zhenyu Zheng <zheng.zhenyu@outlook.com>

bact commented on model/Software/Classes/ContentIdentifier.md:
```diff
@@ -29,3 +29,15 @@ It can be used for verifying its identity and integrity.
   - minCount: 1
   - maxCount: 1
+
+
```
```suggestion

```

bact commented on model/Software/Classes/Package.md:
```diff
@@ -59,3 +59,24 @@ Note that some of these could be represented in SPDX as a file as well.
+* 一个被其他包或文件使用或依赖的独立分发软件(例如 一个Python包, 一个 Go 模块 ...)
+* 一个容器镜像，以及/或者容器镜像里的一个镜像层
+* 一个或多个子包的集合
+* 一个特定时间点的git代码仓库快照
```
```suggestion
- 一个tar,zip或其他压缩文件
- 一个目录或者子目录
- 一个被其他包或文件使用或依赖的独立分发软件(例如 一个Python包, 一个 Go 模块 ...)
- 一个容器镜像，以及/或者容器镜像里的一个镜像层
- 一个或多个子包的集合
- 一个特定时间点的git代码仓库快照
```

Use `-` for bullet (instead of `*`) in all Markdown files in spdx-spec and spec-3-model repos
(for consistency and also to avoid issue of confusing with `*emphasis*` and `**bold**`)

bact commented on model/Software/Properties/copyrightText.md:
```diff
@@ -38,3 +38,23 @@ value as described above.
+
+* SPDX数据创建者没有尝试确定此字段；
+
+* SPDX数据创建者故意没有提供信息（这样做不应推断任何含义）。
```
```suggestion
- SPDX数据创建者已尝试但无法达到合理的客观判断；
- SPDX数据创建者没有尝试确定此字段；
- SPDX数据创建者故意没有提供信息（这样做不应推断任何含义）。
```

bact commented on model/Software/Properties/packageUrl.md:
````diff
@@ -34,3 +34,23 @@ significant on the left to the least significant components on the right.
+scheme:type/namespace/name@version?qualifiers#subpath
+```
+
+每个组件的定义可以在本规范的相应附录([Annex](https://../../../annexes/pkg-url-specification.md))中找到。已知类型的类型定义([type definitions](https://github.com/package-url/purl-spec/blob/b33dda1cf4515efa8eabbbe8e9b140950805f845/PURL-TYPES.rst).)可以在包URL类型定义中找到。
````
```suggestion
每个组件的定义可以在本规范的相应附录([Annex](../../../annexes/pkg-url-specification.md))中找到。
已知类型的类型定义([type definitions](https://github.com/package-url/purl-spec/blob/b33dda1cf4515efa8eabbbe8e9b140950805f845/PURL-TYPES.rst))可以在包URL类型定义中找到。
```

Removing `https://`

bact commented on model/Software/Vocabularies/FileKindType.md:
```diff
@@ -20,3 +20,19 @@ In the future, this may be extended to other kinds (e.g. network based files).
+
+## Entries @zh-Hans
+
+- 文件（file）:文件代表一个单独的文件（默认）。
+
+- 目录（directory）:文件代表一个目录以及存储在该目录中的所有内容。
```
```suggestion
- file: (文件) 文件代表一个单独的文件（默认）。
- directory: (目录) 文件代表一个目录以及存储在该目录中的所有内容。
```

For entry names, should we put the exact entry names first then followed by the translation (if any) in the parentheses instead? This is to emphasis that the actual and expected value is the one displayed the first (outside the parentheses), and the one in parentheses is only for informational purpose.

@zvr @kestewart @goneall @NorioKobota @ZhengZhenyu  What do you think?
Once we agree on this, I will update the [translation guideline](https://github.com/spdx/spdx-3-model/blob/develop/docs/translation.md), as currently there's no clear recommended way to do this. (Currently, the guidelines *implicitly* saying that the entry names shouldn't be translated - by saying *"Only model summaries, descriptions, and vocabulary entry descriptions can be translated. Other content, such as model metadata, cannot be translated."*)

The standardization of this entry name + translated name (in parentheses or else) format is also needed if we want to have the translated entry name in the RDF. So the spec-parser can know where to look, and maybe generate something similar to:

```ttl
<https://spdx.org/rdf/3.0.1/terms/Software/FileKindType/file> a owl:NamedIndividual,
        ns4:FileKindType ;
    rdfs:label "file"@en ;
    rdfs:comment "The file represents a single file (default)."@en ;
    rdfs:label "文件"@zh-Hans ;
    rdfs:comment "文件代表一个单独的文件（默认）。"@zh-Hans .
```

Of course, the normative and only label that we will use in any code will be English. Translations are only for informational purpose.

--

Another way to do this, esp if translation of entry names are not preferred to formally be occurs in the multilingual `rdfs:label`, is to move the translation of entry names to the entry description part (after `:`) instead.

Like this:

```Markdown
- file: (文件) 文件代表一个单独的文件（默认）。
- directory: (目录)  文件代表一个目录以及存储在该目录中的所有内容。
```

This way, there's no need to change the spec-parser and at the same time we can still keep the translated entry names -- as part of the `rdfs:comment` (and not the `rdfs:label`).

```ttl
<https://spdx.org/rdf/3.0.1/terms/Software/FileKindType/file> a owl:NamedIndividual,
        ns4:FileKindType ;
    rdfs:label "file" ;
    rdfs:comment "The file represents a single file (default)."@en ;
    rdfs:comment "(文件) 文件代表一个单独的文件（默认）。"@zh-Hans .
```

--

Also in CJK (Chinese-Japanese-Korean) scripts, they have their own Unicode codepoints for parentheses.
For example, the closing parenthesis in the original text "- 文件（file）:" above is a "Fullwidth Right Parenthesis" (U+FF09) which rendered with a "space" after but logically there is no space character between "）" and ":". We either have to make the parser aware of this, or we have to tell the translator to use an ASCII closing parenthesis (U+0029).

goneall replied:
Whichever approach makes it easier for the spec parser, since the humans will be reading the output of the parser (e.g. the HTML pages).  I believe this would be having the translations follow the colon (`-file: (文件) 文件代表一个单独的文件（默认）。`) - but I'll leave that up to @zvr to decide.

zvr replied:
The entries (things before the `:`) have to stay the same, since they are actually RDF individuals. So it has to be:
```
- file: something
- directory: something
```

If the translation prefers or requires it, the "something" part may start with the translated entry, but let's not mandate it, since I think some languages will not, for example, translate all relationship statuses.

bact replied:
Thanks. I have update the change suggestions above according to @zvr and @goneall comments.

bact reviewed (changes requested):
Few formatting requests and a question about how to represent the translated entry names.

bact committed:
No double blank lines

I will commit trivial minor suggested changes here, so we can focus on the question of entry names

Signed-off-by: Arthit Suriyawongkul <arthit@gmail.com>

bact committed:
Use - (dash) for bullet list (instead of *)

I will commit trivial minor suggested changes here, so we can focus on the question of entry names

Signed-off-by: Arthit Suriyawongkul <arthit@gmail.com>

bact committed:
Remove https:// prefix from local file link

I will commit trivial minor suggested changes here, so we can focus on the question of entry names

Signed-off-by: Arthit Suriyawongkul <arthit@gmail.com>

bact committed:
Use - (dash) for bullet list (instead of *)

I will commit trivial minor suggested changes here, so we can focus on the question of entry names

Signed-off-by: Arthit Suriyawongkul <arthit@gmail.com>

bact commented on model/SimpleLicensing/Properties/licenseExpression.md:
```diff
@@ -26,3 +26,16 @@ license expression as a combination of license objects.
 - name: licenseExpression
 - Nature: DataProperty
 - Range: xsd:string
+
+
```
```suggestion

```

bact commented on model/SimpleLicensing/Properties/customIdToUri.md:
```diff
@@ -29,3 +29,18 @@ CustomLicenseAddition, or SimpleLicensingText.
 - name: customIdToUri
 - Nature: ObjectProperty
 - Range: /Core/DictionaryEntry
+
+
```
```suggestion

```

bact commented on model/Software/Classes/ContentIdentifier.md:
```diff
@@ -29,3 +29,13 @@ It can be used for verifying its identity and integrity.
   - type: xsd:anyURI
   - minCount: 1
   - maxCount: 1
+## Summary @zh-Hans
```
```suggestion

## Summary @zh-Hans
```

One blank line before and after the list/heading.

bact commented on model/Software/Vocabularies/ContentIdentifierType.md:
```diff
@@ -18,3 +18,17 @@ ContentIdentifierType specifies the type of a content identifier.
+## Entries @zh-Hans
+
+- gitoid：[Gitoid](https://www.iana.org/assignments/uri-schemes/prov/gitoid)代表 [Git Object ID](https://git-scm.com/book/en/v2/Git-Internals-Git-Objects)。类型为blob的gitoid是二进制工件的唯一哈希值。gitoid可以代表软件工件的工件标识符（[Artifact Identifier](https://github.com/omnibor/spec/blob/eb1ee5c961c16215eb8709b2975d193a2007a35d/spec/SPEC.md#artifact-identifier-types)），也可以代表软件工件相关联的工件输入清单（ [Artifact Input Manifest](https://git ...
+
```
```suggestion
```

Keep same formatting as the English version

bact commented on model/Software/Vocabularies/SbomType.md:
```diff
@@ -31,3 +31,25 @@ A single SBOM can have multiple SBOM document types associated with it.
+## Summary @zh-Hans
+
+提供一组值，用于描述工具可能创建的常见SBOM（软件物料清单）类型。
+
+
```
```suggestion

```

No double blank lines

bact committed:
No double blank lines

I will commit trivial minor change suggestions here, so we can focus on the question of entry names.

Signed-off-by: Arthit Suriyawongkul <arthit@gmail.com>

bact committed:
No double blank lines

I will commit trivial minor change suggestions here, so we can focus on the question of entry names.

Signed-off-by: Arthit Suriyawongkul <arthit@gmail.com>

bact committed:
One one line before and after the list

I will commit trivial minor change suggestions here, so we can focus on the question of entry names.

Signed-off-by: Arthit Suriyawongkul <arthit@gmail.com>

bact committed:
Keep same formatting as English ver

I will commit trivial minor change suggestions here, so we can focus on the question of entry names.

Signed-off-by: Arthit Suriyawongkul <arthit@gmail.com>

bact committed:
No double blank lines

I will commit trivial minor change suggestions here, so we can focus on the question of entry names.

Signed-off-by: Arthit Suriyawongkul <arthit@gmail.com>

bact commented on model/Software/Vocabularies/SbomType.md:
```diff
@@ -31,3 +31,23 @@ A single SBOM can have multiple SBOM document types associated with it.
+
+## Entries @zh-Hans
+
+- 设计（design）: SBOM是针对预期的、计划中的软件项目或产品，包括了新软件工件的组成部分（其中一些可能尚不存在）。
+- 源代码（source）: 直接从开发环境、源文件和用于构建产品包的依赖关系中创建的SBOM。
+- 构建（build）: 在构建软件的过程中生成的SBOM，以从源文件、依赖关系、构建组件、构建过程临时数据和其他SBOMs等数据中创建一个可发布的工件（例如，可执行文件或包）。
+- 部署态（deployed）: SBOM提供了系统上存在的软件清单。这可能是其他SBOM的组合，结合了配置选项的分析和在（可能是模拟的）部署环境中执行行为的检查。
+- 运行时（runtime）: 通过监控系统运行软件生成的SBOM，以捕获系统中存在的组件以及外部调用或动态加载的组件。在某些情况下，这也可能被称为“监控”或“动态”SBOM。
+- 分析（analyzed）: 在构建工件后，通过对工件（例如，可执行文件、包、容器和虚拟机镜像）进行分析生成的SBOM。这种分析通常需要多种启发式方法。在某些情况下，这也可能被称为“第三方”SBOM。
```
```suggestion
- design: (设计) SBOM是针对预期的、计划中的软件项目或产品，包括了新软件工件的组成部分（其中一些可能尚不存在）。
- source: (源代码) 直接从开发环境、源文件和用于构建产品包的依赖关系中创建的SBOM。
- build: (构建) 在构建软件的过程中生成的SBOM，以从源文件、依赖关系、构建组件、构建过程临时数据和其他SBOMs等数据中创建一个可发布的工件（例如，可执行文件或包）。
- deployed: (部署态) SBOM提供了系统上存在的软件清单。这可能是其他SBOM的组合，结合了配置选项的分析和在（可能是模拟的）部署环境中执行行为的检查。
- runtime: (运行时) 通过监控系统运行软件生成的SBOM，以捕获系统中存在的组件以及外部调用或动态加载的组件。在某些情况下，这也可能被称为“监控”或“动态”SBOM。
- analyzed: (分析) 在构建工件后，通过对工件（例如，可执行文件、包、容器和虚拟机镜像）进行分析生成的SBOM。这种分析通常需要多种启发式方法。在某些情况下，这也可能被称为“第三方”SBOM。
```

Keep the entry name intact. Put the translated entry name in the description part instead.

bact commented on model/Software/Vocabularies/SoftwarePurpose.md:
```diff
@@ -52,3 +52,47 @@ conclusions about the context in which the Element exists.
+
+## Entries @zh-Hans
+
+- 应用程序（application）: 该元素是一个软件应用程序。
+- 归档文件（archive）: 该元素是一个或多个文件的归档集合（如.tar、.zip等）。
+- 物料清单（bom）: 该元素是一个物料清单。
+- 配置数据（configuration）: 该元素是配置数据。
+- 容器（container）: 该元素是一个容器镜像，可以被容器运行时应用程序使用。
+- 数据（data）: 该元素是数据。
+- 设备（device）: 该元素指的是芯片模组、处理器或电路板。
+- 磁盘映像（diskImage）: 该元素指的是可以写入磁盘、在虚拟机中启动等的磁盘映像。磁盘映像通常包含启动所需的大部分或全部组件，如引导加载程序、内核、固件、用户空间等。
+- 设备驱动程序（deviceDriver）: 该元素代表控制硬件设备的软件。
+- 文档（documentation）: 该元素是文档。
+- 证据（evidence）: 该元素是证明规范或要求已得到满足的证据。
+- 可执行文件（executable）: 该元素是可以计算机上运行的工件。
+- 文件（file）: 该元素是一个可以独立分发的单个文件（配置文件、静态链接的二进制文件、Kubernetes部署等）。
+- 文件系统映像（filesystemImage）: 该元素是一个可以写入磁盘（或虚拟）分区的文件系统映像。
+- 固件（firmware）: 该元素提供对设备硬件的低级控制。
+- 框架（framework）: 该元素是一个软件框架。
+- 安装程序（install）: 该元素用于在磁盘上安装软件。
+- 库（library）: 该元素是一个软件库。
+- 清单（manifest）: 该元素是一个软件清单。
+- 模型（model）: 该元素是机器学习或人工智能模型。
+- 模块（module）: 该元素是软件的一个模块。
+- 操作系统（operatingSystem）: 该元素是一个操作系统。
+- 其他（other）: 该元素不属于其他任何类别。
+- 补丁（patch）: 该元素包含一组用于更新、修复或改进另一个元素的更改。
+- 平台（platform）: 该元素代表一个运行时环境。
+- 需求（requirement）: 该元素提供了作为另一个元素输入所需的需求。
+- 源代码（source）: 该元素是单个或一组源代码文件。
+- 规范（specification）: 该元素是一个计划、指南或策略，说明如何创建、执行或分析应用程序。
+- 测试（test）: 该元素是用来验证软件元素功能的测试。
```
```suggestion
- application: (应用程序) 该元素是一个软件应用程序。
- archive: (归档文件) 该元素是一个或多个文件的归档集合（如.tar、.zip等）。
- bom: (物料清单) 该元素是一个物料清单。
- configuration: (配置数据) 该元素是配置数据。
- container: (容器) 该元素是一个容器镜像，可以被容器运行时应用程序使用。
- data: (数据) 该元素是数据。
- device: (设备) 该元素指的是芯片模组、处理器或电路板。
- diskImage: (磁盘映像) 该元素指的是可以写入磁盘、在虚拟机中启动等的磁盘映像。磁盘映像通常包含启动所需的大部分或全部组件，如引导加载程序、内核、固件、用户空间等。
- deviceDriver: (设备驱动程序) 该元素代表控制硬件设备的软件。
- documentation: (文档) 该元素是文档。
- evidence: (证据) 该元素是证明规范或要求已得到满足的证据。
- executable: (可执行文件) 该元素是可以计算机上运行的工件。
- file: (文件) 该元素是一个可以独立分发的单个文件（配置文件、静态链接的二进制文件、Kubernetes部署等）。
- filesystemImage: (文件系统映像) 该元素是一个可以写入磁盘（或虚拟）分区的文件系统映像。
- firmware: (固件) 该元素提供对设备硬件的低级控制。
- framework: (框架) 该元素是一个软件框架。
- install: (安装程序) 该元素用于在磁盘上安装软件。
- library: (库) 该元素是一个软件库。
- manifest: (清单) 该元素是一个软件清单。
- model: (模型) 该元素是机器学习或人工智能模型。
- module: (模块) 该元素是软件的一个模块。
- operatingSystem: (操作系统) 该元素是一个操作系统。
- other: (其他) 该元素不属于其他任何类别。
- patch: (补丁) 该元素包含一组用于更新、修复或改进另一个元素的更改。
- platform: (平台) 该元素代表一个运行时环境。
- requirement: (需求) 该元素提供了作为另一个元素输入所需的需求。
- source: (源代码) 该元素是单个或一组源代码文件。
- specification: (规范) 该元素是一个计划、指南或策略，说明如何创建、执行或分析应用程序。
- test: (测试) 该元素是用来验证软件元素功能的测试。
```

Keep the entry name intact. Put the informative translated entry name in the description part instead.

ZhengZhenyu committed:
Update model/Software/Vocabularies/SoftwarePurpose.md

Co-authored-by: Arthit Suriyawongkul <arthit@gmail.com>
Signed-off-by: Zhenyu Zheng <zheng.zhenyu@outlook.com>

ZhengZhenyu committed:
Update model/Software/Vocabularies/SbomType.md

Co-authored-by: Arthit Suriyawongkul <arthit@gmail.com>
Signed-off-by: Zhenyu Zheng <zheng.zhenyu@outlook.com>

ZhengZhenyu committed:
Update model/Software/Vocabularies/FileKindType.md

Co-authored-by: Arthit Suriyawongkul <arthit@gmail.com>
Signed-off-by: Zhenyu Zheng <zheng.zhenyu@outlook.com>

bact commented on model/SimpleLicensing/Classes/AnyLicenseInfo.md:
```diff
@@ -32,3 +32,22 @@ It can be:
+`AnyLicenseInfo`是一个抽象类，表示由一个或多个许可证（可选包括附加文本）组成的许可证组合，这些许可证可根据[SPDX许可证表达式语法](../../../annexes/spdx-license-expressions.md)进行组合。
+
+`AnyLicenseInfo`可用于软件工件的许可属性。
+
+
```
```suggestion

```

Remove an extra blank line.

bact commented on model/SimpleLicensing/Properties/customIdToUri.md:
```diff
@@ -29,3 +29,16 @@ CustomLicenseAddition, or SimpleLicensingText.
 - name: customIdToUri
 - Nature: ObjectProperty
 - Range: /Core/DictionaryEntry
+## Summary @zh-Hans
```
```suggestion
- Range: /Core/DictionaryEntry

## Summary @zh-Hans
```

Need one blank line before and after the header.

bact commented on model/SimpleLicensing/Properties/licenseExpression.md:
```diff
@@ -26,3 +26,14 @@ license expression as a combination of license objects.
 - name: licenseExpression
 - Nature: DataProperty
 - Range: xsd:string
+## Summary @zh-Hans
```
```suggestion
- Range: xsd:string

## Summary @zh-Hans
```

Need one blank line before and after the header.

bact commented on model/Software/Classes/Package.md:
```diff
@@ -59,3 +59,24 @@ Note that some of these could be represented in SPDX as a file as well.
+
+## Description @zh-Hans
+
+ 包(`Package`)指与软件分发相关的任何内容单元。
```
```suggestion
包(`Package`)指与软件分发相关的任何内容单元。
```

Remove extra space at the beginning.

bact commented on model/Software/Vocabularies/SbomType.md:
```diff
@@ -31,3 +31,23 @@ A single SBOM can have multiple SBOM document types associated with it.
+
+## Summary @zh-Hans
+
+提供一组值，用于描述工具可能创建的常见SBOM（软件物料清单）类型。
+## Description @zh-Hans
```
```suggestion
提供一组值，用于描述工具可能创建的常见SBOM（软件物料清单）类型。

## Description @zh-Hans
```

Need one blank line before and after the header.

bact reviewed (changes requested):
Few formatting suggestions.

bact committed:
Update model/SimpleLicensing/Classes/AnyLicenseInfo.md

Signed-off-by: Arthit Suriyawongkul <arthit@gmail.com>

bact committed:
Update model/SimpleLicensing/Properties/customIdToUri.md

Signed-off-by: Arthit Suriyawongkul <arthit@gmail.com>

bact committed:
Update model/SimpleLicensing/Properties/licenseExpression.md

Signed-off-by: Arthit Suriyawongkul <arthit@gmail.com>

bact committed:
Update model/Software/Classes/Package.md

Signed-off-by: Arthit Suriyawongkul <arthit@gmail.com>

bact committed:
Update model/Software/Vocabularies/SbomType.md

Signed-off-by: Arthit Suriyawongkul <arthit@gmail.com>

bact commented on model/SimpleLicensing/Classes/AnyLicenseInfo.md:
```diff
@@ -32,3 +32,20 @@ It can be:
+
+`AnyLicenseInfo`是一个抽象类，表示由一个或多个许可证（可选包括附加文本）组成的许可证组合，这些许可证可根据[SPDX许可证表达式语法](../../../annexes/spdx-license-expressions.md)进行组合。
+
+`AnyLicenseInfo`可用于软件工件的许可属性。
+它可以是
+- `NoneLicense`；
```
```suggestion
`AnyLicenseInfo`可用于软件工件的许可属性。

它可以是

- `NoneLicense`；
```

Need one blank line before and after the list.

bact committed:
Update model/SimpleLicensing/Classes/AnyLicenseInfo.md

Signed-off-by: Arthit Suriyawongkul <arthit@gmail.com>

gomico commented:
The translations appear accurate and well-crafted. The flow and tone align seamlessly with the original English documents.

helen-ls commented:
Reviewed. The organization is logical, and the language flows naturally. The grammar is solid, and the message is communicated clearly.

bact commented on model/Software/Classes/File.md:
```diff
@@ -34,3 +34,15 @@ a directory and all content stored in that directory.
 - /Core/Element/name
   - minCount: 1
+
+## Summary  @zh-Hans
+
+指计算机上存储内容的任意对象。
+
+## Description  @zh-Hans
```
```suggestion
## Description @zh-Hans
```

bact replied:
```suggestion
## Summary @zh-Hans

指计算机上存储内容的任意对象。

## Description @zh-Hans
```

bact committed:
Remove extra spaces

Signed-off-by: Arthit Suriyawongkul <arthit@gmail.com>

bact commented:
With @gomico and @helen-ls approvals here and with a decision at https://github.com/spdx/spdx-3-model/pull/949#issuecomment-2715654416, I will merge this.
